qiskit/dagcircuit/dagcircuit.py

```diff
@@ -1393,6 +1393,34 @@ def substitute_node(self, node, op, inplace=False):
             self._decrement_op(node.op)
         return new_node
 
+    def separable_circuits(self) -> List["DAGCircuit"]:
+        """Separate the dag circuit into its weakly connected components.
+
+           The global phase information in `self` will not be maintained in the
+           subcircuits returned by this method.
+        """
+        connected_components = rx.weakly_connected_components(self._multi_graph)
+
+        # Collect each disconnected subgraph
+        disconnected_subgraphs = []
+        for components in connected_components:
+            disconnected_subgraphs.append(self._multi_graph.subgraph(list(components)))
+
+        # Create new DAGCircuit objects from each of the rustworkx subgraph objects
+        decomposed_dags = []
+        for subgraph in disconnected_subgraphs:
+            new_dag = self.copy_empty_like()
+            for node in subgraph.nodes():
```

Comment:
I'm not sure if this is guaranteed to return the nodes in the correct order. We are making that assumption by popping them onto the end of the wire in order.

@kdk @jakelishman ?

Reply:
`DAGCircuit.nodes()` is implementation-defined iteration order (in practice, I think it'll shake out to be node insertion order if there's no removals). You possibly want to iterate through `DAGCircuit.topological_op_nodes()` - that'll skip the terminators for the bits (they're already in from `copy_empty_like`) and then do the topological sort.

Reply:
`subgraph` is the `rustworkx.PyDAG.subgraph` object, so it's not clear to me how to use`DAGCircuit.topological_op_nodes` here. These loops exists to create those `DAGCircuit` objects. Is there an easier way to instantiate them directly from the subgraphs?

Reply:
Oh my bad, I read it wrong.  I imagine you want to call

```python
def key(node):
    node.sort_key

rustworkx.lexicographical_topological_sort(subgraph, key=key)
```

but @mtreinish can correct me.  I don't know what all the rustworkx functions / PyDAG methods are.